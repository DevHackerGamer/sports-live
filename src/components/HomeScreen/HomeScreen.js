import React from 'react';

const HomeScreen = ({ setActiveTab, setSelectedLeague, leagues, latestNews, newsLoading }) => {
  const features = [
    { 
      title: "Live Matches", 
      description: "Follow games in real-time", 
      icon: "⚽",
      action: () => setActiveTab('matches')
    },
    { 
      title: "Players", 
      description: "Player stats and profiles", 
      icon: "👤",
      action: () => setActiveTab('players')
    },
    { 
      title: "Standings", 
      description: "League tables and rankings", 
      icon: "🏆",
      action: () => setActiveTab('leagueStandings')
    },
    { 
      title: "Favorites", 
      description: "Your followed content", 
      icon: "❤️",
      action: () => setActiveTab('favorites')
    }
  ];

  return (
    <div className="home-screen">
      {/* Hero Section */}
      <section className="hero">
        <div className="hero-content">
          <h1>Experience Football <span>Like Never Before</span></h1>
          <p>Real-time scores, in-depth statistics, and coverage of all major leagues.</p>
        </div>
      </section>

      {/* Main Dashboard Grid */}
      <div className="dashboard-grid">
        {/* News Sidebar - Left */}
        <section className="news-sidebar">
          <div className="news-header">
            <h2>Latest News</h2>
            <button 
              className="view-all-btn"
              onClick={() => setActiveTab('news')}
            >
              View All
            </button>
          </div>
          
          <div className="news-feed">
            {newsLoading ? (
              <div className="news-loading">
                <div className="loading-spinner"></div>
                <p>Loading news...</p>
              </div>
            ) : latestNews.length > 0 ? (
              latestNews.map((article, index) => (
                <div 
                  key={article._id || index} 
                  className="news-item"
                  onClick={() => setActiveTab('news')}
                >
                  {article.images?.[0] && (
                    <div className="news-image">
                      <img src={article.images[0].url} alt={article.headline} />
                    </div>
                  )}
                  <div className="news-content">
                    <h4>{article.headline}</h4>
                    <p className="news-excerpt">
                      {article.description?.substring(0, 100)}...
                    </p>
                    <div className="news-meta">
                      <span className="news-time">
                        {new Date(article.published).toLocaleDateString()}
                      </span>
                    </div>
                  </div>
                </div>
              ))
            ) : (
              <div className="no-news">
                <p>No recent news available</p>
                <button 
                  className="retry-btn"
                  onClick={() => window.location.reload()}
                >
                  Refresh
                </button>
              </div>
            )}
          </div>
        </section>

        {/* Right Column */}
        <div className="dashboard-right-column">
          {/* Quick Actions */}
          <section className="quick-actions">
            <h2>Quick Actions</h2>
            <div className="actions-grid">
              {features.map((f, idx) => (
                <div className="action-card" key={idx} onClick={f.action}>
                  <div className="action-icon">{f.icon}</div>
                  <h3>{f.title}</h3>
                  <p>{f.description}</p>
                </div>
              ))}
            </div>
          </section>

          {/* Featured Leagues */}
          <section className="featured-leagues">
            <h2>Featured Leagues</h2>
            <div className="featured-leagues-grid">
              {leagues.slice(0, 6).map((league, idx) => (
                <div 
                  className="featured-league-card" 
                  key={idx}
                  onClick={() => {
                    setSelectedLeague(league.key);
                    setActiveTab('leagueStandings');
                  }}
                >
                  <div className="featured-league-inner">
                    <div className="featured-league-front">
                      <div className="featured-league-logo">
                        <img src={league.img} alt={league.name} />
                      </div>
                    </div>
                    <div 
                      className="featured-league-back" 
                      style={{ backgroundColor: league.color }}
                    >
                      <div className="featured-league-back-content">
                        <h3>{league.name}</h3>
                        <p>{league.desc}</p>
                        <button className="btn btn-small">
                          View League
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              ))}
            </div>
          </section>
        </div>
      </div>
    </div>
  );
};

export default HomeScreen;