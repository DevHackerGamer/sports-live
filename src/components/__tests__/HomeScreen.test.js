import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import HomeScreen from '../HomeScreen/HomeScreen';

// Mock FontAwesome icons
jest.mock('@fortawesome/react-fontawesome', () => ({
  FontAwesomeIcon: ({ icon, ...props }) => <span data-testid="font-awesome-icon" {...props} />,
}));

describe('HomeScreen Component', () => {
  const mockSetActiveTab = jest.fn();
  const mockSetSelectedLeague = jest.fn();

  const defaultProps = {
    setActiveTab: mockSetActiveTab,
    setSelectedLeague: mockSetSelectedLeague,
    leagues: [
      { key: 'PL', name: 'Premier League', img: '/pl-logo.png', color: '#3D195B', desc: 'English top division' },
      { key: 'LL', name: 'La Liga', img: '/laliga-logo.png', color: '#FF0000', desc: 'Spanish top division' },
      { key: 'SA', name: 'Serie A', img: '/seriea-logo.png', color: '#008C45', desc: 'Italian top division' },
      { key: 'BL', name: 'Bundesliga', img: '/bundesliga-logo.png', color: '#D20026', desc: 'German top division' },
      { key: 'FL', name: 'Ligue 1', img: '/ligue1-logo.png', color: '#091C3E', desc: 'French top division' },
      { key: 'CL', name: 'Champions League', img: '/ucl-logo.png', color: '#0F4C81', desc: 'European competition' },
    ],
    latestNews: [
      {
        _id: '1',
        headline: 'Breaking News: Team Wins Championship',
        description: 'In an incredible match, the team secured their victory in the final minutes.',
        published: '2024-01-15T10:00:00Z',
        images: [{ url: '/news1.jpg', alt: 'Championship win' }]
      },
      {
        _id: '2',
        headline: 'Transfer Window Update',
        description: 'Major transfers happening across European leagues as deadline approaches.',
        published: '2024-01-14T15:30:00Z',
        images: []
      }
    ],
    newsLoading: false,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders hero section correctly', () => {
    render(<HomeScreen {...defaultProps} />);

    expect(screen.getByText('Experience Football')).toBeInTheDocument();
    expect(screen.getByText('Like Never Before')).toBeInTheDocument();
    expect(screen.getByText(/Real-time scores, in-depth statistics/i)).toBeInTheDocument();
  });

  it('renders news sidebar with latest news', () => {
    render(<HomeScreen {...defaultProps} />);

    expect(screen.getByText('Latest News')).toBeInTheDocument();
    expect(screen.getByText('View All')).toBeInTheDocument();
    expect(screen.getByText('Breaking News: Team Wins Championship')).toBeInTheDocument();
    expect(screen.getByText('Transfer Window Update')).toBeInTheDocument();
  });

  

  
  it('shows no news message when news array is empty', () => {
    render(<HomeScreen {...defaultProps} latestNews={[]} />);

    expect(screen.getByText('No recent news available')).toBeInTheDocument();
  });

  it('renders quick actions section with all features', () => {
    render(<HomeScreen {...defaultProps} />);

    expect(screen.getByText('Quick Actions')).toBeInTheDocument();
    expect(screen.getByText('Live Matches')).toBeInTheDocument();
    expect(screen.getByText('Follow games in real-time')).toBeInTheDocument();
    expect(screen.getByText('Highlights')).toBeInTheDocument();
    expect(screen.getByText('Watch match highlights')).toBeInTheDocument();
    expect(screen.getByText('Standings')).toBeInTheDocument();
    expect(screen.getByText('League tables and rankings')).toBeInTheDocument();
    expect(screen.getByText('Favorites')).toBeInTheDocument();
    expect(screen.getByText('Your followed content')).toBeInTheDocument();
  });

  it('renders featured leagues section', () => {
    render(<HomeScreen {...defaultProps} />);

    expect(screen.getByText('Featured Leagues')).toBeInTheDocument();
    expect(screen.getByText('Premier League')).toBeInTheDocument();
    expect(screen.getByText('La Liga')).toBeInTheDocument();
    expect(screen.getByText('Serie A')).toBeInTheDocument();
    expect(screen.getByText('Bundesliga')).toBeInTheDocument();
    expect(screen.getByText('Ligue 1')).toBeInTheDocument();
    expect(screen.getByText('Champions League')).toBeInTheDocument();
  });

  it('renders league logos', () => {
    render(<HomeScreen {...defaultProps} />);

    const leagueLogos = screen.getAllByRole('img');
    // First image is news, rest are league logos
    expect(leagueLogos[1]).toHaveAttribute('src', '/pl-logo.png');
    expect(leagueLogos[2]).toHaveAttribute('src', '/laliga-logo.png');
  });

  it('calls setActiveTab when quick action cards are clicked', () => {
    render(<HomeScreen {...defaultProps} />);

    // Test Live Matches action
    fireEvent.click(screen.getByText('Live Matches'));
    expect(mockSetActiveTab).toHaveBeenCalledWith('matches');

    // Test Highlights action
    fireEvent.click(screen.getByText('Highlights'));
    expect(mockSetActiveTab).toHaveBeenCalledWith('highlights');

    // Test Standings action
    fireEvent.click(screen.getByText('Standings'));
    expect(mockSetActiveTab).toHaveBeenCalledWith('leagueStandings');

    // Test Favorites action
    fireEvent.click(screen.getByText('Favorites'));
    expect(mockSetActiveTab).toHaveBeenCalledWith('favorites');
  });

  it('calls setActiveTab when View All news button is clicked', () => {
    render(<HomeScreen {...defaultProps} />);

    fireEvent.click(screen.getByText('View All'));
    expect(mockSetActiveTab).toHaveBeenCalledWith('news');
  });

  it('calls setActiveTab when news items are clicked', () => {
    render(<HomeScreen {...defaultProps} />);

    const firstNewsItem = screen.getByText('Breaking News: Team Wins Championship');
    fireEvent.click(firstNewsItem);
    expect(mockSetActiveTab).toHaveBeenCalledWith('news');
  });

  it('calls setSelectedLeague and setActiveTab when league cards are clicked', () => {
    render(<HomeScreen {...defaultProps} />);

    // Click on Premier League card
    fireEvent.click(screen.getByText('Premier League'));
    expect(mockSetSelectedLeague).toHaveBeenCalledWith('PL');
    expect(mockSetActiveTab).toHaveBeenCalledWith('leagueStandings');

    // Click on La Liga card
    fireEvent.click(screen.getByText('La Liga'));
    expect(mockSetSelectedLeague).toHaveBeenCalledWith('LL');
    expect(mockSetActiveTab).toHaveBeenCalledWith('leagueStandings');
  });

  it('limits featured leagues to first 6', () => {
    const manyLeagues = [
      { key: 'PL', name: 'Premier League', img: '/pl.png', color: '#000', desc: 'Desc' },
      { key: 'LL', name: 'La Liga', img: '/ll.png', color: '#000', desc: 'Desc' },
      { key: 'SA', name: 'Serie A', img: '/sa.png', color: '#000', desc: 'Desc' },
      { key: 'BL', name: 'Bundesliga', img: '/bl.png', color: '#000', desc: 'Desc' },
      { key: 'FL', name: 'Ligue 1', img: '/fl.png', color: '#000', desc: 'Desc' },
      { key: 'CL', name: 'Champions League', img: '/cl.png', color: '#000', desc: 'Desc' },
      { key: 'EL', name: 'Europa League', img: '/el.png', color: '#000', desc: 'Desc' }, // 7th league
    ];

    render(<HomeScreen {...defaultProps} leagues={manyLeagues} />);

    expect(screen.getByText('Premier League')).toBeInTheDocument();
    expect(screen.getByText('La Liga')).toBeInTheDocument();
    expect(screen.getByText('Serie A')).toBeInTheDocument();
    expect(screen.getByText('Bundesliga')).toBeInTheDocument();
    expect(screen.getByText('Ligue 1')).toBeInTheDocument();
    expect(screen.getByText('Champions League')).toBeInTheDocument();
    expect(screen.queryByText('Europa League')).not.toBeInTheDocument();
  });

  it('formats news dates correctly', () => {
    render(<HomeScreen {...defaultProps} />);

    // Check that dates are formatted (format will depend on locale)
    const dateElements = screen.getAllByText(/2024/);
    expect(dateElements.length).toBeGreaterThan(0);
  });

  it('truncates news descriptions', () => {
    render(<HomeScreen {...defaultProps} />);

    const excerpt = screen.getByText(/In an incredible match, the team secured their victory in the final minutes\.\.\./);
    expect(excerpt).toBeInTheDocument();
  });

  it('handles news items without images', () => {
    render(<HomeScreen {...defaultProps} />);

    // The second news item has no images, but should still render
    expect(screen.getByText('Transfer Window Update')).toBeInTheDocument();
  });

  it('matches snapshot with default props', () => {
    const { container } = render(<HomeScreen {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with loading state', () => {
    const { container } = render(<HomeScreen {...defaultProps} newsLoading={true} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with empty news', () => {
    const { container } = render(<HomeScreen {...defaultProps} latestNews={[]} />);
    expect(container).toMatchSnapshot();
  });
});