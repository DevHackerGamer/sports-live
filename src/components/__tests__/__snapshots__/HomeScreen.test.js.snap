// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Component matches snapshot with default props 1`] = `
<div>
  <div
    class="home-screen"
  >
    <section
      class="hero"
    >
      <div
        class="hero-content"
      >
        <h1>
          Experience Football 
          <span>
            Like Never Before
          </span>
        </h1>
        <p>
          Real-time scores, in-depth statistics, and coverage of all major leagues.
        </p>
      </div>
    </section>
    <div
      class="dashboard-grid"
    >
      <section
        class="news-sidebar"
      >
        <div
          class="news-header"
        >
          <h2>
            <span
              data-testid="font-awesome-icon"
              style="margin-right: 0.5rem;"
            />
            Latest News
          </h2>
          <button
            class="view-all-btn"
          >
            View All
          </button>
        </div>
        <div
          class="news-feed"
        >
          <div
            class="news-item"
          >
            <div
              class="news-image"
            >
              <img
                alt="Breaking News: Team Wins Championship"
                src="/news1.jpg"
              />
            </div>
            <div
              class="news-content"
            >
              <h4>
                Breaking News: Team Wins Championship
              </h4>
              <p
                class="news-excerpt"
              >
                In an incredible match, the team secured their victory in the final minutes.
                ...
              </p>
              <div
                class="news-meta"
              >
                <span
                  class="news-time"
                >
                  1/15/2024
                </span>
              </div>
            </div>
          </div>
          <div
            class="news-item"
          >
            <div
              class="news-content"
            >
              <h4>
                Transfer Window Update
              </h4>
              <p
                class="news-excerpt"
              >
                Major transfers happening across European leagues as deadline approaches.
                ...
              </p>
              <div
                class="news-meta"
              >
                <span
                  class="news-time"
                >
                  1/14/2024
                </span>
              </div>
            </div>
          </div>
        </div>
      </section>
      <div
        class="dashboard-right-column"
      >
        <section
          class="quick-actions"
        >
          <h2>
            Quick Actions
          </h2>
          <div
            class="actions-grid"
          >
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Live Matches
              </h3>
              <p>
                Follow games in real-time
              </p>
            </div>
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Highlights
              </h3>
              <p>
                Watch match highlights
              </p>
            </div>
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Standings
              </h3>
              <p>
                League tables and rankings
              </p>
            </div>
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Favorites
              </h3>
              <p>
                Your followed content
              </p>
            </div>
          </div>
        </section>
        <section
          class="featured-leagues"
        >
          <h2>
            Featured Leagues
          </h2>
          <div
            class="featured-leagues-grid"
          >
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Premier League"
                      src="/pl-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(61, 25, 91);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Premier League
                    </h3>
                    <p>
                      English top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="La Liga"
                      src="/laliga-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(255, 0, 0);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      La Liga
                    </h3>
                    <p>
                      Spanish top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Serie A"
                      src="/seriea-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(0, 140, 69);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Serie A
                    </h3>
                    <p>
                      Italian top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Bundesliga"
                      src="/bundesliga-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(210, 0, 38);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Bundesliga
                    </h3>
                    <p>
                      German top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Ligue 1"
                      src="/ligue1-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(9, 28, 62);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Ligue 1
                    </h3>
                    <p>
                      French top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Champions League"
                      src="/ucl-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(15, 76, 129);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Champions League
                    </h3>
                    <p>
                      European competition
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`HomeScreen Component matches snapshot with empty news 1`] = `
<div>
  <div
    class="home-screen"
  >
    <section
      class="hero"
    >
      <div
        class="hero-content"
      >
        <h1>
          Experience Football 
          <span>
            Like Never Before
          </span>
        </h1>
        <p>
          Real-time scores, in-depth statistics, and coverage of all major leagues.
        </p>
      </div>
    </section>
    <div
      class="dashboard-grid"
    >
      <section
        class="news-sidebar"
      >
        <div
          class="news-header"
        >
          <h2>
            <span
              data-testid="font-awesome-icon"
              style="margin-right: 0.5rem;"
            />
            Latest News
          </h2>
          <button
            class="view-all-btn"
          >
            View All
          </button>
        </div>
        <div
          class="news-feed"
        >
          <div
            class="no-news"
          >
            <p>
              No recent news available
            </p>
          </div>
        </div>
      </section>
      <div
        class="dashboard-right-column"
      >
        <section
          class="quick-actions"
        >
          <h2>
            Quick Actions
          </h2>
          <div
            class="actions-grid"
          >
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Live Matches
              </h3>
              <p>
                Follow games in real-time
              </p>
            </div>
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Highlights
              </h3>
              <p>
                Watch match highlights
              </p>
            </div>
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Standings
              </h3>
              <p>
                League tables and rankings
              </p>
            </div>
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Favorites
              </h3>
              <p>
                Your followed content
              </p>
            </div>
          </div>
        </section>
        <section
          class="featured-leagues"
        >
          <h2>
            Featured Leagues
          </h2>
          <div
            class="featured-leagues-grid"
          >
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Premier League"
                      src="/pl-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(61, 25, 91);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Premier League
                    </h3>
                    <p>
                      English top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="La Liga"
                      src="/laliga-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(255, 0, 0);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      La Liga
                    </h3>
                    <p>
                      Spanish top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Serie A"
                      src="/seriea-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(0, 140, 69);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Serie A
                    </h3>
                    <p>
                      Italian top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Bundesliga"
                      src="/bundesliga-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(210, 0, 38);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Bundesliga
                    </h3>
                    <p>
                      German top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Ligue 1"
                      src="/ligue1-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(9, 28, 62);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Ligue 1
                    </h3>
                    <p>
                      French top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Champions League"
                      src="/ucl-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(15, 76, 129);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Champions League
                    </h3>
                    <p>
                      European competition
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`HomeScreen Component matches snapshot with loading state 1`] = `
<div>
  <div
    class="home-screen"
  >
    <section
      class="hero"
    >
      <div
        class="hero-content"
      >
        <h1>
          Experience Football 
          <span>
            Like Never Before
          </span>
        </h1>
        <p>
          Real-time scores, in-depth statistics, and coverage of all major leagues.
        </p>
      </div>
    </section>
    <div
      class="dashboard-grid"
    >
      <section
        class="news-sidebar"
      >
        <div
          class="news-header"
        >
          <h2>
            <span
              data-testid="font-awesome-icon"
              style="margin-right: 0.5rem;"
            />
            Latest News
          </h2>
          <button
            class="view-all-btn"
          >
            View All
          </button>
        </div>
        <div
          class="news-feed"
        >
          <div
            class="news-loading"
          >
            <div
              class="loading-spinner"
            />
            <p>
              Loading news...
            </p>
          </div>
        </div>
      </section>
      <div
        class="dashboard-right-column"
      >
        <section
          class="quick-actions"
        >
          <h2>
            Quick Actions
          </h2>
          <div
            class="actions-grid"
          >
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Live Matches
              </h3>
              <p>
                Follow games in real-time
              </p>
            </div>
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Highlights
              </h3>
              <p>
                Watch match highlights
              </p>
            </div>
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Standings
              </h3>
              <p>
                League tables and rankings
              </p>
            </div>
            <div
              class="action-card"
            >
              <div
                class="action-icon"
              >
                <span
                  data-testid="font-awesome-icon"
                />
              </div>
              <h3>
                Favorites
              </h3>
              <p>
                Your followed content
              </p>
            </div>
          </div>
        </section>
        <section
          class="featured-leagues"
        >
          <h2>
            Featured Leagues
          </h2>
          <div
            class="featured-leagues-grid"
          >
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Premier League"
                      src="/pl-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(61, 25, 91);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Premier League
                    </h3>
                    <p>
                      English top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="La Liga"
                      src="/laliga-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(255, 0, 0);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      La Liga
                    </h3>
                    <p>
                      Spanish top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Serie A"
                      src="/seriea-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(0, 140, 69);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Serie A
                    </h3>
                    <p>
                      Italian top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Bundesliga"
                      src="/bundesliga-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(210, 0, 38);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Bundesliga
                    </h3>
                    <p>
                      German top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Ligue 1"
                      src="/ligue1-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(9, 28, 62);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Ligue 1
                    </h3>
                    <p>
                      French top division
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="featured-league-card"
            >
              <div
                class="featured-league-inner"
              >
                <div
                  class="featured-league-front"
                >
                  <div
                    class="featured-league-logo"
                  >
                    <img
                      alt="Champions League"
                      src="/ucl-logo.png"
                    />
                  </div>
                </div>
                <div
                  class="featured-league-back"
                  style="background-color: rgb(15, 76, 129);"
                >
                  <div
                    class="featured-league-back-content"
                  >
                    <h3>
                      Champions League
                    </h3>
                    <p>
                      European competition
                    </p>
                    <button
                      class="btn btn-small"
                    >
                      View League
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
